### ISSUE TITLE
need to press twice the power button to start SP6

### ISSUE THREAD

**Issue by knard** [linux-surface/linux-surface#373]
since some times now I have to press twice the power button to start my SP6. At the beginning I thought it was a hardware issue but the startup process seems too regular to be hardware related.
- pressing power button the first time and the windows logo appear for 1 second and the laptop shutdown
- pressing a second time will show the windows log for half a second, the screen will turn black for 2 or 3 seconds and the log while show again and the laptop will boot normally.

After some research it's seems to be related to performance management but I have no clue if it's really that and if there is a way to fix it from linux.

### Environment

- Hardware model: SP6
- Kernel version: 5.10.2-arch1-1-surface
- Distribution: Archlinux

**Comment by qzed** [comment]
I have a similar problem. As mentioned it in #344 (which seems to be an entirely different problem), I have no clue what's causing this, or even how to reproduce this reliably on my SB2. May have something to do with the EC.

Any chance you're using a Surface Dock?

**Comment by knard** [comment]
yes I'm using the surface dock. After you mentioned it I have some tests and I can't reproduce it any more. The test sequence was: 
- The laptop was plugged
- I have unplugged it
- I have shutdown the computer
- I have restarted it (needed to press twice)
- I have shut it down while on gdm login screen
- I have restarted it (start at first press)
- I have login into gnome-shell
- I have shutdown the computer
- new restart without issue

Since then I have plugged again the dock, and everything seems fine, every start is working at first push.... very strange.

**Comment by qzed** [comment]
Hmm, I have the feeling that this is somehow related to the dock, but I can't really confirm that. I just have the feeling that, for me at least, this happens only when I shut down with the dock attached or boot with the dock attached. As said earlier, I can't reproduce it reliably and most of the time it works with the first press, so I can't say for certain that the dock is causing this.

Could for example also be some shutdown command (currently unknown to us) that needs to be sent to the EC but is missing.

**Comment by knard** [comment]
from what I saw on Microsoft forums the issue also occur on win10. Seems that disabling the fast boot feature in win10 has solved the problem for some users, some other have reported that setting back the default energy profile has helped.

I remember that I have tested the energy profile that I have tested the profile patch you have provided in the past ( linux-surface/surface-aggregator-module#18) I wanted to switch to a different profile to see if it has any impact, but the rqst.py script doesn't work anymore. Do you know how I can switch the profile now ?

**Comment by qzed** [comment]
With the surface kernel (or alternatively the surface aggregator module installed) you should be able to use the `surface` tool (`surface-control` package, run `surface perf set <number>`) or use the sysfs interface (which the tool uses) directly (see https://github.com/linux-surface/surface-aggregator-module/wiki/Performance-Modes).
